pyp2spec/license_processor.py

```diff
@@ -0,0 +1,146 @@
+from importlib.resources import open_text
+import json
+import re
+import sys
+
+import click
+import requests
+
+
+class TroveConverter:
+    """Takes care of package license classifiers conversion to the Fedora SPDX format.
+    """
+
+    def __init__(self, classifiers):
+        self.classifiers = classifiers
+        self.spdx_identifiers = self.convert_to_spdx()
+
+    def load_trove2fedora_map(self):
+        """Load the file with Python trove classifiers mapped to SPDX identifiers."""
+
+        with open_text("pyp2spec", "classifiers_to_fedora.json") as map_data:
+            return json.load(map_data)
+
+    def convert_to_spdx(self):
+        """Return the list of SPDX identifiers converted from the license classifiers.
+
+        If the conversion of any of the classifiers is not possible, return None.
+        If the source data don't contain such classifier, print this fact to stdout
+        and end the script.
+        """
+
+        trove2fedora_map = self.load_trove2fedora_map()
+        spdx_identifiers = []
+        for classifier in self.classifiers:
+            try:
+                fedora_identifier = trove2fedora_map[classifier]
+            except KeyError:
+                err_string = """{0}: Such classifier doesn't exist.
+                If you believe that's pyp2spec's error, open an issue at project's GitHub repo."""
+                click.secho(err_string.format(classifier), fg="red")
+                sys.exit(1)
```

Comment:
Style note (and if the rest of the app does it this way, feel free to disregard that here and let's talk about it later). I consider methods and functions calling exit a code smell. Raising an exception (other than SystemExit, e.g. ValueError here) and handling exit from the main app entry point is IMHO much cleaner. I don't know if this is just my view or some generally agreed-upon dogrma :)

Reply:
I removed all the `sys.exit`s from license_processor.